src/opt.rs

```diff
@@ -224,20 +227,40 @@ pub struct CommonData {
     pub allow_insecure_host: Option<Vec<String>>,
 }
 
+fn compute_s3_config<M>(s3_options: &M) -> HashMap<String, s3_middleware::S3Config>
```

Comment:
tbh this should go into rattler_config (or rattler_networking). i copied this code way too many times

Reply:
yes, but I didn't want to introduce a dependency on rattler_networking yet. I think we can migrate this code at some point, but probably to `rattler_networking` with an optional dependency on `rattler_config`.